python_scripts/generate_test_data.py

```diff
@@ -0,0 +1,458 @@
+
+        # Get a cursor
+        cursor = db_conn.cursor()
+
+        # Call stored procedure with parameters
+        args = [system_name, system_run_id, destination_barcode, destination_coordinate, source_barcode, source_coordinate]
+        cursor.callproc('updateDestinationPlateWellWithSource', args)
+
+        # commit changes
+        db_conn.commit()
+    except Exception as e:
+        print(e)
+    else:
+        print("Updated destination well %s with picked source %s coordinate %s" % (destination_coordinate, source_barcode, source_coordinate))
+    finally:
+        cursor.close()
+        db_conn.close()
+
+# Update the run record at the end of the run
+def update_run_record(system_name, system_run_id, new_state):
+    try:
+        # open DB connection
+        db_conn = create_database_connection()
+
+        # Get a cursor
+        cursor = db_conn.cursor()
+
+        # Call stored procedure with parameters
+        args = [system_name, system_run_id, new_state]
+        cursor.callproc('updateRunState', args)
+
+        # commit changes
+        db_conn.commit()
+    except Exception as e:
+        print(e)
+    else:
+        print("Updated run record id %s with state %s" % (system_run_id, new_state))
+    finally:
+        cursor.close()
+        db_conn.close()
+
+# Fetch the last run id used
+def get_last_run_id() -> int:
+    last_run_id = 0
+    try:
+        # open DB connection
+        db_conn = create_database_connection()
+
+        # Get a cursor
+        cursor = db_conn.cursor()
+
+        sql = "SELECT MAX(system_run_id) FROM %s.automation_system_runs;" % DB_NAME
+        cursor.execute(sql)
+
+        result = cursor.fetchone()
+        if result[0] is not None:
+            last_run_id = result[0]
+
+    except Exception as e:
+        print(e)
+    else:
+        print("Fetched last run id %s" % last_run_id)
+    finally:
+        cursor.close()
+        db_conn.close()
+        return last_run_id
+
+# Insert a run event record
+def create_run_event_record(system_name, system_run_id, event_type, event):
+    try:
+        # open DB connection
+        db_conn = create_database_connection()
+
+        # Get a cursor
+        cursor = db_conn.cursor()
+
+        # Call stored procedure with parameters
+        args = [system_name, system_run_id, event_type, event]
+        cursor.callproc('createRunEventRecord', args)
+
+        # commit changes
+        db_conn.commit()
+    except Exception as e:
+        print(e)
+    else:
+        print("Created run event type %s with event %s" % (event_type, event))
+    finally:
+        cursor.close()
+        db_conn.close()
+
+# =============
+# Script starts here
+# =============
+
+partial_source_barcode = ''
+sample_uuid_index = 1
+for dest_index in range(NUMBER_DESTINATION_PLATES_TO_MAKE):
+    print("Processing for destination index %s" % dest_index)
+    print("Fetching system configuration")
+#   [SP] Fetch config for CPA
```

Comment:
I would move comments indentation to the same column as the code, in case python gets picky about when a block has finished

Reply:
Ok